files-csharp/DifferentWaysToOverwiteAFile/DifferentWaysToOverwiteAFile.Test/OverwriteExistingFileUnitTests.cs

```diff
@@ -0,0 +1,147 @@
+using DifferentWaysToOverwiteAFile.Examples;
+using System.Text;
+
+namespace DifferentWaysToOverwiteAFile.Test
+{
+    public class OverwriteExistingFileTests
+    {
+
+        [Test]
+        public void UseWriteAllText_ShouldOverwriteTheFile()
+        {
+            var filePath = "content.txt";
+            try
+            {
+                //Arrange
+                var originalContent = "Hello, World!";
+                var newContent = "Hello, Code Maze!";
+
+                // Create the test file and write the original content
+                File.WriteAllText(filePath, originalContent);
+
+                //Act
+                WriteAllText.OverwiteFile(filePath, newContent);
+                var actualContent = File.ReadAllText(filePath);
+
+                //Assert
+                Assert.That(actualContent, Is.EqualTo(newContent));
+            }
+            finally
+            {
+                // Clean up - delete the test file
+                File.Delete(filePath);
+            }
+        }
+
+        [Test]
+        public void UseWriteAllBytes_ShouldOverwriteTheFile()
+        {
+            var filePath = "content.txt";
+            try
+            {
+                //Arrange
+                var originalContent = "Hello, World!";
+                var newContent = "Hello, Code Maze!";
+                var newBytes = Encoding.UTF8.GetBytes(newContent);
+
+                // Create the test file and write the original content
+                File.WriteAllText(filePath, originalContent);
+
+                //Act
+                WriteAllBytes.OverwiteFile(filePath, newBytes);
+                var actualContent = File.ReadAllText(filePath);
+
+                //Assert
+                Assert.That(actualContent, Is.EqualTo(newContent));
```

Comment:
Since the focus of this test is the writing of bytes to the file, rather than writing a string that we get the encoding, we should focus on writing bytes. We can assert that the current content of the file does not match the bytes we are about to write, and then we write the bytes, overwriting the file, read back the bytes and finally assert that the bytes we read back match the bytes we wrote.